boards/nordic/nrf54h20dk/Kconfig.nrf54h20dk

```diff
@@ -2,15 +2,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_NRF54H20DK
-	select SOC_NRF54H20_ENGB_CPUAPP if BOARD_NRF54H20DK_NRF54H20_CPUAPP && BOARD_REVISION = "0.8.0"
```

Comment:
The commit that removes the board revision should come first, so that Kconfig symbols related to the EngB revision are removed when they are not referenced anymore.